### PR TITLE
Sdks/typescript testing

### LINKED ISSUE #171: Add proper  testing to typescript sdk
Typescript SDK has no testing and with it getting more complex, an issue was introduced in 1.1.7. It should have automated tests before new functionality is added.

### PR DESCRIPTION
**Pull request by rvowles**
# Description

Repository & Analytics testing for typescript SDK. Polling & Test parts are very http centric and hard to test.

Fixes #171

### CODE REVIEW

File: sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      logEvent: function(action: string, other: Map<string, string>,
+                         featureStateAtCurrentTime: Array<FeatureStateHolder>) {
+        firedAction = action;
+        firedOther = other;
+        firedFeatures = featureStateAtCurrentTime;
+      }
+    });
+  });
+
+  it('Should enable us to log an event with no other and no features', () => {
```
Comment (IrinaSouth) [inline sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts:32]: with no other?

File: sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    repo.logAnalyticsEvent('name');
+    expect(firedFeatures.length).to.eq(0);
+    expect(firedAction).to.eq('name');
+    // tslint:disable-next-line:no-unused-expression
+    expect(firedOther).to.be.undefined;
+  });
+
+  it('should carry through the other field', () => {
```
Comment (IrinaSouth) [inline sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts:40]: what are we testing here? just not sure of this description

File: sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    const other = new Map();
+    other.set('ga', 'value');
+    repo.logAnalyticsEvent('name', other);
+    expect(firedFeatures.length).to.eq(0);
+    expect(firedAction).to.eq('name');
+    // tslint:disable-next-line:no-unused-expression
+    expect(firedOther).to.eq(other);
+  });
+
+  it('should snapshot the features', () => {
```
Comment (IrinaSouth) [inline sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts:50]: and here

File: sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts (hunk 1 above)
Comment (rvowles) [reply sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts:32]: parameter is called "other"

File: sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts (hunk 2 above)
Comment (rvowles) [reply sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts:40]: the field "other"

**Review by IrinaSouth — APPROVED** [review]

**Pull request merged**